### PR TITLE
shim: support bundle cache level 2

### PR DESCRIPTION
**Pull request by jiazhiguang**
1. shim: support bundle cache level 2
2. epm: fix a bug that it may lose some ancestor paths when building the cache directory
3. shim/signature: add an new interface to get the public key from the signature server
4. shim/vendor: update the module name of shim and update the vendor packages

### CODE REVIEW

**Comment by jiazhang0** [comment]
@jiazhiguang Plz add CI/CD as the part of your work. Thanks.

**Review by jiazhang0 — CHANGES_REQUESTED** [review]
Plz remove the empty line between fix line and sob line.

File: shim/dist/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PROJECT_DIR := $(shell cd ../..; pwd)
 RPMBUILD_DIR := $(shell mktemp -u /tmp/rpmbuild.XXXX)
-RELEASE_TARBALL_URL := https://github.com/alibaba/inclavare-containers/archive/v$(INCLAVARE_CONTAINERS_VERSION).tar.gz
+RELEASE_TARBALL_URL := https://github.com/inclavare-containers/archive/v$(INCLAVARE_CONTAINERS_VERSION).tar.gz
```
Comment (haosanzi) [inline shim/dist/Makefile:3]: `https://github.com/inclavare-containers/archive/v%{version}tar.gz` is not a valid url.

File: shim/dist/deb/build.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ PACKAGE=shim-rune
 PROJECT=inclavare-containers
 VERSION=$(cd ../../..; cat ./VERSION)
 RELEASE_TARBALL=$DEBBUILD_DIR/v$VERSION.tar.gz
-RELEASE_TARBALL_URL=https://github.com/alibaba/inclavare-containers/archive/v$VERSION.tar.gz
+RELEASE_TARBALL_URL=https://github.com/inclavare-containers/archive/v$VERSION.tar.gz
 TARBALL_NAME=$PACKAGE\_$VERSION.orig.tar.gz
```
Comment (haosanzi) [inline shim/dist/deb/build.sh:11]: https://github.com/inclavare-containers/archive/v%{version}tar.gz is not a valid url.

File: shim/dist/deb/debian/control [hunk 3]
```diff
@@ -4,7 +4,7 @@ Priority: extra
 Maintainer: shirong <shirong@linux.alibaba.com>
 Build-Depends: debhelper (>=9)
 Standards-Version: 3.9.8
-Homepage: https://github.com/alibaba/inclavare-containers
```
Comment (haosanzi) [inline shim/dist/deb/debian/control:7]: `https://github.com/inclavare-containers` is not a URL for our program.

File: shim/dist/rpm/shim-rune.spec [hunk 4]
```diff
@@ -13,8 +13,8 @@ Release: %{centos_base_release}%{?dist}
 Summary: shim for Inclavare Containers(runE)
 Group: Development/Tools
 License: Apache License 2.0
-URL: https://github.com/alibaba/%{PROJECT}
-Source0: https://github.com/alibaba/%{PROJECT}/archive/v%{version}.tar.gz
+URL: https://github.com%{PROJECT}
+Source0: https://github.com%{PROJECT}/archive/v%{version}.tar.gz
```
Comment (haosanzi) [inline shim/dist/rpm/shim-rune.spec:18]: `https://github.com/inclavare-containers` is not a URL for our program.
`https://github.com/inclavare-containers/archive/v%{version}tar.gz` is not the real package url.

**Review by hustliyilin — APPROVED** [review]

**Pull request closed without merging**